### PR TITLE
test: mock database interface

### PR DESCRIPTION
**Pull request by beanc16**
🔗 Linked issue
resolves https://github.com/unjs/db0/issues/123

❓ Type of change

* [ ]  📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
* [ ]  🐞 Bug fix (a non-breaking change that fixes an issue)
* [ ]  👌 Enhancement (improving an existing functionality like performance)
* [ ] ✨ New feature (a non-breaking change that adds functionality)
* [x]  🧹 Chore (updates to the build process or auxiliary tools and libraries)
* [ ]  ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
* [ ]  📚 Description
In this PR I've implemented database mocks in unit tests.

📝 Checklist

* [x]  I have linked an issue or discussion.
* [ ]  I have updated the documentation accordingly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,6 +59,7 @@
     "@vitest/coverage-v8": "^2.1.4",
     "automd": "^0.3.12",
     "better-sqlite3": "^11.5.0",
+    "bun": "^1.1.34",
```
Comment (beanc16) [inline package.json:62]: `bun` being necessary for some tests wasn't listed in documentation, so I added it here for posterity. Feel free to include that `npm i -g bun` is necessary in documentation instead if we don't want it as a dependency.

File: test/connectors/_tests.ts [hunk 2]
```diff
@@ -1,47 +1,54 @@
...
-          "lastName": "Doe",
-        },
-      ]
-    `);
+    expect(rows).toEqual([{
+      id: userId,
+      firstName: "John",
+      lastName: "Doe",
+      email: "john.doe@example.com",
+    }]);
```
Comment (pi0) [inline test/connectors/_tests.ts:52]: This is nice change in tests but seems unrelated can you please extract to separate PR? 🙏🏼

**Comment by pi0** [comment]
Thanks for the PR. I'm trying to understand benefits and what extra (real) coverage it is adding.

It seems replacing actual functionality of databases with our "expected" behavior (which can vary from real behavior).

> which appears to limit the ability to enable wider test coverage due to connection requirements.

A simple in-mem sqlite connection would enable us to run any kind of unit tests (that are not testing connectors but other functionalities) unless I'm wrong?

**Comment by beanc16** [comment]
> Thanks for the PR. I'm trying to understand benefits and what extra (real) coverage it is adding.
> 
> It seems replacing actual functionality of databases with our "expected" behavior (which can vary from real behavior).

For me, it's moreso about the distinction between unit vs integration tests. What's here prior to this PR works great as integration tests, since they connect directly to the database. Though, unit tests are inherently meant to test something and get the same result every time with no external dependencies (like a database connection).

What sparked my PR was that I was kind of surprised that I wouldn't be able to mock or run tests in any way in an isolated manner without having all types of database connections set up first. This PR makes it so you can test changes without a database connection and make sure things work (assuming this is something we want). Though, if we want integration tests to ensure connections actually work 100% of the time, I'd think we want unit vs integration tests to be separate so it's clear as to what type of test is what.

**Comment by pi0** [comment]
Thanks for explanation. I see value of unit tests but what are we exactly going to "unit" test with this PR?

**Comment by beanc16** [comment]
@pi0 Sorry for the delay, just came back to this. I went back through this again, and I think I misunderstood some of the project's setup. Apologies!
